pkg/istio-agent/agent.go

```diff
@@ -205,6 +210,35 @@ func NewAgent(proxyConfig *mesh.ProxyConfig, agentOpts *AgentOptions, sopts *sec
 		cfg:         agentOpts,
 		secOpts:     sopts,
 		envoyOpts:   eopts,
+		watcher:     filewatcher.NewWatcher(),
+		tlsOpts:     &citadel.TLSOptions{},
+	}
+}
+
+func (a *Agent) watcherHandler(ctx context.Context) {
+	for {
+		select {
+		case gotEvent := <-a.watcher.Events(a.tlsOpts.RootCert):
+			log.Infof("Receive file %s event %v", a.tlsOpts.RootCert, gotEvent)
+			if a.caClient != nil {
+				err := a.caClient.InitClient()
```

Comment:
Not strictly related to this PR but seems like `InitClient` has data races as there is no locking at all.. Do we have sufficient test coverage?

Reply:
I'll review all things about data race.

Reply:
after reviewed the entire flow, it's unnecessary to `InitClient`.